GUI/Types/Renderer/WorldLoader.cs

```diff
@@ -274,7 +274,7 @@ private void LoadEntitiesFromLump(Scene scene, LoadResult result, EntityLump ent
 
                 if (animation != default)
                 {
-                    modelNode.LoadAnimation(animation); // Load only this animation
+                    modelNode.LoadAnimations(guiContext.FileLoader);
```

Comment:
Won't this load every single animation file unnecessarily, which this was designed to avoid. Have you compared this to master? Dota map might not be a good enough test, would need a map that has models with many animations.

Reply:
Is it actually worth it, especially given that code path is unused right now?

It loaded all embedded animations either way, and if animation wasn't present there - it loaded all referenced animation groups until needed animation was found, so in the worst case (missing animation or it's present in the last file) it'd load all referenced animations as well.

The worst is that it only added reference animations from file that contained needed animation, so if that path triggered multiple times for ref animations - it loaded them over and over.
https://github.com/SteamDatabase/ValveResourceFormat/blob/f06c48d3f48d959b3a79ae43e7226a39e27d4cb1/GUI/Types/Renderer/ModelSceneNode.cs#L246-L250

Reply:
I tested on a2_drainage, looks like it's not a problem.